tsl/src/continuous_aggs/create.c

```diff
@@ -1116,39 +1123,110 @@ cagg_validate_query(const Query *query, const bool finalized, const char *cagg_s
 				 detail->len > 0 ? errdetail("%s", detail->data) : 0));
 	}
 
-	/* finalized cagg doesn't have those restrictions anymore */
+	/* Finalized cagg doesn't have those restrictions anymore */
 	if (!finalized)
 	{
 		/* validate aggregates allowed */
 		cagg_agg_validate((Node *) query->targetList, NULL);
 		cagg_agg_validate((Node *) query->havingQual, NULL);
 	}
-
+	/* Check if there are only two tables in the from list */
 	fromList = query->jointree->fromlist;
-	if (list_length(fromList) != 1 || !IsA(linitial(fromList), RangeTblRef))
+	if (list_length(fromList) > 2 )
 	{
 		ereport(ERROR,
 				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-				 errmsg("only one hypertable allowed in continuous aggregate view")));
+				 errmsg("only one hypertable and one normal table are"
+				 		" allowed in continuous aggregate view")));
 	}
-	/* check if we have a hypertable in the FROM clause */
-	rtref = linitial_node(RangeTblRef, query->jointree->fromlist);
-	rte = list_nth(query->rtable, rtref->rtindex - 1);
+	/* Extra checks for joins in Caggs */
+	if (list_length(fromList) == 2)
+	{
+		rtref = linitial_node(RangeTblRef, query->jointree->fromlist);
+		rte = list_nth(query->rtable, rtref->rtindex - 1);
+		rtref2 = lsecond_node(RangeTblRef, query->jointree->fromlist);
+		rte2 = list_nth(query->rtable, rtref2->rtindex - 1);
 
-	/* FROM only <tablename> sets rte->inh to false */
-	if ((rte->relkind != RELKIND_RELATION && rte->relkind != RELKIND_VIEW) || rte->tablesample ||
-		rte->inh == false)
+		/*
+		 * If one table in from clause is cagg on cagg then other needs to be a
```

Comment:
I think this comment is confusing, right? If one table is cagg on cagg then it doesn't matter if the other is a normal table if we don't support cagg on cagg in joins. Or will support for that be added in a follow-up PR?

Reply:
For now joins with cagg on cagg is not supported, we can work it on for the follow-up PR.